### PR TITLE
Add a test for early return in the resource selection algorithm

### PR DESCRIPTION
**Pull request by muodov**
This adds a test for https://github.com/whatwg/html/issues/3065

### CODE REVIEW

**Review by annevk — COMMENTED** [review]
Thanks for writing this!

Do any browsers fail this? Are there bug reports?

File: html/semantics/embedded-content/media-elements/loading-the-media-resource/resource-selection-resumes-onload.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<!doctype html>
+<title>resource selection should not delay the load event indefinitely</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<div id=log></div>
+<video></video>
+<script>
+async_test(function(t) {
+  var v = document.querySelector('video');
+  v.onloadstart = t.step_func(function() { assert_unreached(); });
```
[the remark this answers is not in the document]
Comment (muodov) [reply html/semantics/embedded-content/media-elements/loading-the-media-resource/resource-selection-resumes-onload.html]: done, thanks for the hint!

File: html/semantics/embedded-content/media-elements/loading-the-media-resource/resource-selection-resumes-onload.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<!doctype html>
+<title>resource selection should not delay the load event indefinitely</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<div id=log></div>
+<video></video>
+<script>
+async_test(function(t) {
+  var v = document.querySelector('video');
```
[the remark this answers is not in the document]
Comment (muodov) [reply html/semantics/embedded-content/media-elements/loading-the-media-resource/resource-selection-resumes-onload.html]: Changed. Initially I just copypasted from another resource selection test. Are there specific guidelines regarding ES6 syntax in WPT?

**Comment by muodov** [comment]
> Do any browsers fail this? Are there bug reports?

I checked it in the latest Safari, Chrome and Firefox on macOS - all green.
In IE11 it raises exceptions, but as far as I can see it's from the testharness itself, so I assume it's not supposed to be supported?

File: html/semantics/embedded-content/media-elements/loading-the-media-resource/resource-selection-resumes-onload.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (annevk) [reply html/semantics/embedded-content/media-elements/loading-the-media-resource/resource-selection-resumes-onload.html]: Not particularly, but you can use what's in all browsers generally and using newer features can help with readability.

**Review by annevk — APPROVED** [review]

**Pull request merged**